### DISCUSSION TITLE
How to query channels.editAdmin using TdLib .NET C#?

### DISCUSSION THREAD

**Discussion by AndreiTsapurin** [egramtel/tdsharp#211] [Q&A]
I have a .NET C# project. The TdLib nuget package is connected to it.
Please tell me how I can execute the request: https://core.telegram.org/method/channels.editAdmin using this library?
To be honest, I don’t understand at all how to use this documentation. I couldn’t find any examples for .NET C#.

If I understand correctly, there are several options for executing queries:

 
```
//Using strongly typed APIs example
    using TdLib;
    
    var client = new TdClient();
    TdApi.Ok ok = await client.ExecuteAsync(new TdApi.SetAuthenticationPhoneNumber
    {
        PhoneNumber = phoneNumber
    });
```

And

 
```
//Using json client
    using TdLib;
    
    var json = ""; // json data
    double timeout = 1.0; // 1 second
    
    using (var jsonClient = new TdJsonClient())
    {
        jsonClient.Send(json); // send request
        var result = jsonClient.Receive(timeout); // receive response
    }
```

The actual question is: how to execute the channels.editAdmin request using the first or second option?

UPD:

I found an example code for calling this method in JS:

 
```
// make the bot an admin of the channel
            const makeAdmin: Api.TypeUpdates = await client.invoke(
              new Api.channels.EditAdmin({
                channel: new Api.InputChannel({
                  channelId: channelId as bigInt.BigInteger,
                  accessHash: accessHash as bigInt.BigInteger,
                }),
                userId: '@BotNameHere',
                adminRights: new Api.ChatAdminRights({
                  changeInfo: true,
                  postMessages: true,
                  editMessages: true,
                  deleteMessages: true,
                  banUsers: true,
                  inviteUsers: true,
                  pinMessages: true,
                  addAdmins: true,
                  anonymous: false,
                  manageCall: true,
                  other: false,
                }),
                rank: 'Administrator',
              })
            );
            if (makeAdmin instanceof Api.Updates) {
              logger.info('Bot is now an admin of the channel');
            }
          }
```

But it’s still unclear how to adapt it to my situation.

**Comment by ForNeVeR** [comment]
Sadly, I have no idea what is this function.

This project so far provides bindings for [`td_api.tl`](https://github.com/tdlib/td/blob/f35dea776cdaa8b986e2a634dfabf0dafe659be7/td/generate/scheme/td_api.tl). You can see that there's no such function there.

I see that this function is seemingly defined in [`telegram_api.tl`](https://github.com/tdlib/td/blob/f35dea776cdaa8b986e2a634dfabf0dafe659be7/td/generate/scheme/telegram_api.tl), but I don't understand what's the use of that file.

If you know what JSON you want to send, then use `TdJsonClient`.

If you understand what's the use of the mysterious `telegram_api.tl`, then please explain and I'm ready to improve the project by incorporating that.

Otherwise, sadly I don't know any of these either.

**Reply by AndreiTsapurin** [reply]
I'm trying to match the methods available in the TdLib library and those provided in the Telegram Api documentation: https://core.telegram.org/api. I see that this is not the same thing. Is there any relations between your library and this documentation?

**Reply by ForNeVeR** [reply]
This is not a question for us. We only provide whatever's available in tdlib. What _is_ available in tdlib and how it maps onto the documentation is a question for tdlib.
